nvflare/fuel/hci/zip_utils.py

```diff
@@ -165,25 +157,38 @@ def convert_legacy_zip(zip_data: bytes) -> bytes:
         The converted zip data
     """
 
+    meta: Optional[dict] = None
     reader = io.BytesIO(zip_data)
     with ZipFile(reader, "r") as in_zip:
         info_list = in_zip.infolist()
         folder_name = info_list[0].filename.split("/")[0]
         meta_path = _path_join(folder_name, META_NAME)
         if next((info for info in info_list if info.filename == meta_path), None):
             # Already in job layout
-            return zip_data
+            meta_data = in_zip.read(meta_path)
+            meta = json.loads(meta_data)
+            if JobMetaKey.JOB_FOLDER_NAME not in meta:
+                meta[JobMetaKey.JOB_FOLDER_NAME.value] = folder_name
+            else:
+                return zip_data
 
```

Comment:
1. if meta.json exist already, 
    you insert `JobMetaKey.JOB_FOLDER_NAME` if it is not there.
    
  This logic is fine.